OpenRA.Game/Graphics/PlayerColorRemap.cs

```diff
@@ -26,12 +25,16 @@ public static int GetRemapIndex(int[] ramp, int i)
 			return ramp[i];
 		}
 
-		public PlayerColorRemap(int[] ramp, HSLColor c, float rampFraction)
+		public PlayerColorRemap(int[] ramp, Color c, float rampFraction)
 		{
+			var h = c.GetHue() / 360.0f;
+			var s = c.GetSaturation();
+			var l = c.GetBrightness();
+
 			// Increase luminosity if required to represent the full ramp
-			var rampRange = (byte)((1 - rampFraction) * c.L);
-			var c1 = new HSLColor(c.H, c.S, Math.Max(rampRange, c.L)).RGB;
-			var c2 = new HSLColor(c.H, c.S, (byte)Math.Max(0, c.L - rampRange)).RGB;
+			var rampRange = (byte)((1 - rampFraction) * l);
+			var c1 = Color.FromAhsl(h, s, Math.Max(rampRange, l));
```

Comment:
FromAhsl/FromAhsv should take an alpha value, or be named FromHsl/FromHsv

Reply:
This is modeled after `System.Drawing.Color.FromArgb`, which provides non-alpha overloads.  I can change this and our clone of `FromArgb` if you insist, but i'd prefer not to.